### PR TITLE
docs(core): implement RTL service documentation

### PR DESCRIPTION
**Pull request by khotcholava**
closes [#11854](https://github.com/SAP/fundamental-ngx/issues/11854)

## Description
Added documentation for using RtlService to implement RTL in applications. Included examples and guidance on how and when to use RtlService, and added an RTL section to each control's documentation.

### CODE REVIEW

**Comment by droshev** [comment]
the switch you add is interfering with the other one:
Screenshot 2024-07-30 at 09 15 19

**Review by InnaAtanasova — COMMENTED** [review]
Screenshot 2024-07-30 at 9 35 05 AM
 why do you need 2 switches? When the top one is used, the bottom one seems broken.
I think it would make more sense to use as an example one of the languages that are mentioned above.

**Comment by alexhristov14** [comment]
RtlService docs should be in CDK utilities instead of Core since the service is in CDK.

File: libs/docs/core/rtl-service/rtl-service-docs.component.html [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    <li>Hebrew (he)</li>
+    <li>Khowar (khw)</li>
+    <li>Kashmiri (ks)</li>
+    <li>Kurdish (ku)</li>
+    <li>Pashto (ps)</li>
+    <li>Urdu (ur)</li>
+    <li>Yiddish (yi)</li>
+</ul>
+<description>
+    You can override this list by providing a custom list view the <code>RTL_LANGUAGE</code> injection token.
```
Comment (mikerodonnell89) [inline libs/docs/core/rtl-service/rtl-service-docs.component.html:32]:
```suggestion
    You can override this list by providing a custom list to the <code>RTL_LANGUAGE</code> injection token.
```

File: libs/docs/core/rtl-service/examples/rtl-service-basic-example/rtl-service-basic-example.component.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    // Signal to track RTL state
+    rtl$ = computed(() => !!this._rtlService?.rtlSignal());
+    // Signal to track a direction
+    _dir$ = computed<Direction>(() => {
+        return this.rtl$() ? 'rtl' : 'ltr';
+    });
+
+    constructor() {
+        effect(() => {
+            this._updateTextContainerDirection();
```
Comment (mikerodonnell89) [inline libs/docs/core/rtl-service/examples/rtl-service-basic-example/rtl-service-basic-example.component.ts:36]: I am new to the signals API (still doing most of my work in Angular 15), so correct me if wrong but doesn't an effect need to have a reference to a signal inside of it? What is the purpose of having "effect" here?

File: libs/docs/core/rtl-service/rtl-service-docs.component.html [hunk 3]
```diff
@@ -0,0 +1,38 @@
+<fd-docs-section-title id="rtl-service-usage" componentName="rtl-service"> RTL Service Usage </fd-docs-section-title>
+<description>
+    The <code>RtlService</code> service can be used to manage the Right-To-Left (RTL) settings in your application.
+</description>
+
+<description>
+    It uses a <code>BehaviorSubject</code> to track the RTL status and provides a <code>Signal</code> wrapper for
+    reactive programming. The service can be configured with a custom list of RTL languages using the
```
Comment (mikerodonnell89) [inline libs/docs/core/rtl-service/rtl-service-docs.component.html:8]: An example of this would be cool too

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
Looking good! Couple comments above

File: libs/docs/core/rtl-service/examples/rtl-service-basic-example/rtl-service-basic-example.component.ts (hunk 2 above)
Comment (mikerodonnell89) [inline libs/docs/core/rtl-service/examples/rtl-service-basic-example/rtl-service-basic-example.component.ts:36]: I am new to signals so forgive me (still doing most of my work in Angular 15). I am confused why there would be an effect here, what are you demonstrating?

Comment (khotcholava) [reply libs/docs/core/rtl-service/examples/rtl-service-basic-example/rtl-service-basic-example.component.ts:36]: If you look at the `_updateTextContainerDirection` function, it uses a signal within the function. I'm not directly using the signals in the effect; instead, `_updateTextContainerDirection` handles it. When the `_dir$()` signal changes, the effect will work.

File: libs/docs/core/rtl-service/rtl-service-docs.component.html (hunk 3 above)
Comment (khotcholava) [reply libs/docs/core/rtl-service/rtl-service-docs.component.html:8]: Added example.

File: libs/docs/core/rtl-service/examples/rtl-service-basic-example/rtl-service-basic-example.component.ts (hunk 2 above)
Comment (mikerodonnell89) [reply libs/docs/core/rtl-service/examples/rtl-service-basic-example/rtl-service-basic-example.component.ts]: So doesn't clicking the toggle button call `_updateTextContainerDirection()`, which triggers the effect, which calls `_updateTextContainerDirection()` again? Is that intended?

Comment (khotcholava) [reply libs/docs/core/rtl-service/examples/rtl-service-basic-example/rtl-service-basic-example.component.ts]: You're absolutely correct. There's no need to call `_updateTextContainerDirection` explicitly on button click. The effect will automatically trigger whenever `_dir$()` changes.

**Review by mikerodonnell89 — APPROVED** [review]

**Review by dpavlenishvili — APPROVED** [review]

File: package.json [hunk 4]
```diff
@@ -1,177 +1,178 @@
...
-      "yarn nx format:write --uncommitted"
-    ],
-    "*.{js,ts,html,md,json,yml}": [
-      "yarn eslint --fix"
-    ],
-    "*.properties": [
-      "yarn nx run i18n:transform-translations"
-    ]
-  }
+    "name": "fundamental-ngx",
```
Comment (droshev) [inline package.json:2]: why did you change this?
